dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -25,18 +25,48 @@ Criteria::Criteria(const boost::shared_ptr<criteria::CriteriaParams>& params) :
 
 Criteria::~Criteria() {}
 
+void
+Criteria::printAllFailingCriteriaIntoLog(std::vector<std::shared_ptr<FailingCriteria> > failingCriteriaVec,
+                                          const boost::shared_ptr<timeline::Timeline>& timeline,
+                                          double currentTime) const {
+  std::sort(failingCriteriaVec.begin(),
+            failingCriteriaVec.end(),
+            [](const std::shared_ptr<FailingCriteria>& failingcrit1, const std::shared_ptr<FailingCriteria>& failingcrit2) {
+                return failingcrit1->getDistance() > failingcrit2->getDistance();
+            });
+
+  for (size_t logIndex = 0; logIndex < failingCriteriaVec.size(); ++logIndex) {
```

Comment:
for (size_t logIndex = 0; logIndex < failingCriteriaVec.size(); ++logIndex) {
  failingCriteriaVec[logIndex]->printOneFailingCriteriaIntoLog();
  if (timeline && logIndex < maxFailingCritArraySize)
    failingCriteriaVec[logIndex]->printOneFailingCriteriaIntoTimeline(timeline, currentTime);
}

Reply:
If we implement this, the if condition will be checked a large number of times.